dnf5-plugins/config-manager_plugin/addrepo.cpp

```diff
@@ -0,0 +1,519 @@
+                        M_("Error in added repository configuration file. Cannot set repository option \"{}={}\": {}"),
+                        key,
+                        key_val,
+                        std::string{ex.what()});
+                }
+            }
+        }
+    } catch (const Error & ex) {
+        std::error_code ec;
+        std::filesystem::remove(tmpfilepath, ec);
+        throw;
+    }
+
+    // All tests passed. Renames the configuration file to the final name.
+    std::filesystem::rename(tmpfilepath, dest_path);
+    logger->info("config-manager: Added repofile \"{}\" from \"{}\"", dest_path.string(), source_repofile.location);
+    set_file_permissions(dest_path);
+}
+
+
+void ConfigManagerAddRepoCommand::create_repo(
+    std::string repo_id,
+    const std::map<std::string, std::string> & repo_opts,
+    const std::filesystem::path & dest_repo_dir) {
+    auto & ctx = get_context();
+    auto & base = ctx.base;
+    auto logger = base.get_logger();
+
+    // Test for presence of required arguments.
+    // And sets the URL - used to create the repository ID if not specified.
+    std::string url;
+    if (const auto it = repo_opts.find("baseurl"); it != repo_opts.end() && !it->second.empty()) {
+        const auto urls = OptionStringList(std::vector<std::string>{}).from_string(it->second);
+        if (urls.empty() || (url = urls.front()).empty()) {
+            throw ConfigManagerError(M_("Bad baseurl: {}={}"), it->first, it->second);
+        }
+    } else if (const auto it = repo_opts.find("mirrorlist"); it != repo_opts.end() && !it->second.empty()) {
+        url = it->second;
+    } else if (const auto it = repo_opts.find("metalink"); it != repo_opts.end() && !it->second.empty()) {
+        url = it->second;
+    } else {
+        throw cli::ArgumentParserMissingDependentArgumentError(
+            M_("One of --from-repofile=<URL>, --set=baseurl=<URL>, --set=mirrorlist=<URL>, --set=metalink=<URL> "
+               "must be set to a non-empty URL"));
+    }
+
+    if (repo_id.empty()) {
+        repo_id = sanitize_url(url);
+    }
+
+    if (save_filename.empty()) {
+        save_filename = repo_id;
+    }
+    if (!save_filename.ends_with(".repo")) {
+        save_filename += ".repo";
+    }
+    auto dest_path = dest_repo_dir / save_filename;
+
+    test_if_filepath_not_exist(dest_path, true);
+    test_if_ids_not_already_exist({repo_id}, dest_path);
+
+    ConfigParser parser;
+
+    if (file_policy == FilePolicy::ADD_OR_REPLACE && std::filesystem::exists(dest_path)) {
+        parser.read(dest_path);
+        if (parser.has_section(repo_id)) {
+            // If the repository with the id already exists, it will be removed.
+            parser.remove_section(repo_id);
+        }
+    }
+
+    parser.add_section(repo_id);
+
+    // Sets the default repository name. May be overwritten with "--set=name=<name>".
+    parser.set_value(repo_id, "name", "created by dnf5 config-manager");
```

Comment:
What about to include repoid in the name? Proposing something like `<RepoID> created by dnf5 config-manager`. Keeping like it is is also OK, therefore the change is optional.